pertpy/tools/_coda/_base_coda.py

```diff
@@ -2173,6 +2177,11 @@ def plot_effects_umap(  # pragma: no cover # noqa: D417
         data_coda = mdata[modality_key_2]
         if isinstance(effect_name, str):
             effect_name = [effect_name]
+        if isinstance(palette, Colormap):
+            palette = {
+                cluster: palette(i % palette.N)
+                for i, cluster in enumerate(data_rna.obs[cluster_key].unqiue().to_list())
```

Comment:
```suggestion
                for i, cluster in enumerate(data_rna.obs[cluster_key].unique().to_list())
```

given that this typo was in the code, do your example cases actually cover this?

Reply:
Good question. I will double-check.😅

Reply:
It was not as it is in `plot_effects_umap` which I failed to get working without setting a `palette`.

Reply:
Right, I see. Then this is on my plate :)